credentials/tls/certprovider/pemfile/builder_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package pemfile
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+type testConfig struct {
+	CertificateFile   string `json:"certificate_file,omitempty"`
+	PrivateKeyFile    string `json:"private_key_file,omitempty"`
+	CACertificateFile string `json:"ca_certificate_file,omitempty"`
+	RefreshInterval   string `json:"refresh_interval,omitempty"`
+}
+
+func makeJSONConfig(t *testing.T, cfg testConfig) json.RawMessage {
```

Comment:
Is this better than putting the literal JSON into the test?  It's better to avoid logic in tests unless it is worth it.

Reply:
Done.